Source/Core/Core/NetPlayServer.cpp

```diff
@@ -968,6 +1011,17 @@ bool NetPlayServer::RequestStartGame()
 
     m_start_pending = true;
   }
+
```

Comment:
The structure of these statements is all messed up. The way it's currently written, the game will launch immediately if save sync is enabled but cheat sync is disabled

I suggest the following change to fix this:
```diff
@@ -1011,6 +1011,10 @@
 
     m_start_pending = true;
   }
+  else
+  {
+    return true;
+  }
 
   // Check To Send Gecko Codes to Clients
   if (m_settings.m_SyncGeckoCodes && m_players.size() > 1)
@@ -1021,12 +1025,12 @@
       return false;
     }
   }
-
   else
   {
-    return StartGame();
+    return true;
   }
 
+  StartGame();
   return true;
 }
```

Reply:
i made a change to this before you raised the issue. let me know if you think it should still be changed.

https://github.com/dolphin-emu/dolphin/blob/51793fddadad8e0f0aa1f651353b42a9b841863f/Source/Core/Core/NetPlayServer.cpp#L1008

Reply:
Now the game will fail to start with only one player if either save sync or cheat sync is enabled. How about this?

```diff
@@ -1007,20 +1007,24 @@
 // called from ---GUI--- thread
 bool NetPlayServer::RequestStartGame()
 {
+  bool start_now = true;
+
   if (m_settings.m_SyncSaveData && m_players.size() > 1)
   {
+    start_now = false;
+    m_start_pending = true;
     if (!SyncSaveData())
     {
       PanicAlertT("Error synchronizing save data!");
+      m_start_pending = false;
       return false;
     }
-
-    m_start_pending = true;
   }
 
   // Check To Send Codes to Clients
   if (m_settings.m_SyncCodes && m_players.size() > 1)
   {
+    start_now = false;
     m_start_pending = true;
     if (!SyncCodes())
     {
@@ -1028,13 +1032,10 @@
       m_start_pending = false;
       return false;
     }
-
   }
 
-  else if (!m_settings.m_SyncSaveData && !m_settings.m_SyncCodes)
-  {
+  if (start_now)
     return StartGame();
-  }
 
   return true;
 }
```

Reply:
you got it